sub-packages/bionemo-scdl/src/bionemo/scdl/io/single_cell_memmap_dataset.py

```diff
@@ -523,6 +522,27 @@ def regular_load_h5ad(
         # Store the row idx array
         self.row_index[0 : num_rows + 1] = count_data.indptr.astype(int)
 
+        if self.include_obs:
+            self.obs = {}
+            os.makedirs(f"{self.data_path}/obs", exist_ok=True)
+            for obs_name in adata.obs:
+                if adata.obs[obs_name].dtype == "category":
+                    self.obs[obs_name] = np.memmap(
+                        f"{str(Path(self.data_path).absolute())}/obs/{obs_name}",
+                        dtype="S20",
+                        shape=(num_rows + 1,),
+                        mode=Mode.CREATE_APPEND,
+                    )
+
+                else:
+                    self.obs[obs_name] = np.memmap(
+                        f"{str(Path(self.data_path).absolute())}/obs/{obs_name}",
+                        dtype=adata.obs[obs_name].dtype,
+                        shape=(num_rows + 1,),
+                        mode=Mode.CREATE_APPEND,
+                    )
```

Comment:
```suggestion
                if adata.obs[obs_name].dtype == "category":
                    obs_dtype = "S20"
                else:
                    obs_dtype = adata.obs[obs_name].dtype
                self.obs[obs_name] = np.memmap(
                        f"{str(Path(self.data_path).absolute())}/obs/{obs_name}",
                        dtype=obs_dtype,
                        shape=(num_rows + 1,),
                        mode=Mode.CREATE_APPEND,
                    )
```
Maybe somethign like the above since it looks like the commands are the same other than setting the dtype?